src/modules/Sticky/Sticky.js

```diff
@@ -255,33 +266,34 @@ export default class Sticky extends Component {
   }
 
   stickToContextBottom = (e) => {
-    const top = this.contextRect.bottom - this.stickyRect.height
-
     _.invoke(this.props, 'onBottom', e, this.props)
 
     this.stick(e)
-    this.setState({ top, bottom: null })
+    this.bound(true)
     this.pushing(true)
   }
 
   stickToContextTop = (e) => {
     _.invoke(this.props, 'onTop', e, this.props)
 
     this.unstick(e)
+    this.bound(false)
     this.pushing(false)
   }
 
   stickToScreenBottom = (e) => {
     const { bottomOffset: bottom } = this.props
 
     this.stick(e)
+    this.bound(false)
     this.setState({ bottom, top: null })
```

Comment:
I'm not sure that we really need the `bound()` method, this case shows it because we have duplicate call of `setState()` there